src/main/java/hudson/plugins/throttleconcurrents/pipeline/ThrottleStep.java

```diff
@@ -0,0 +1,81 @@
+package hudson.plugins.throttleconcurrents.pipeline;
+
+import hudson.Extension;
+import hudson.model.TaskListener;
+import hudson.plugins.throttleconcurrents.ThrottleJobProperty;
+import hudson.util.FormValidation;
+import org.jenkinsci.plugins.workflow.steps.Step;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.StepDescriptor;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+import javax.annotation.Nonnull;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.StringTokenizer;
+
+public class ThrottleStep extends Step implements Serializable {
+    private String categories;
+
+    @DataBoundConstructor
+    public ThrottleStep(@Nonnull String categories) {
+        this.categories = categories;
+    }
+
+    @Nonnull
+    public String getCategories() {
+        return categories;
+    }
+
+    @Nonnull
+    public List<String> getCategoriesList() {
+        List<String> catList = new ArrayList<>();
+
+        StringTokenizer tokenizer = new StringTokenizer(categories, ",");
+        while (tokenizer.hasMoreTokens()) {
+            catList.add(tokenizer.nextToken().trim());
+        }
+
+        return catList;
+    }
+
+    @Override
+    public StepExecution start(StepContext context) throws Exception {
+        return new ThrottleStepExecution(this, context);
+    }
+
+    private static final long serialVersionUID = 1L;
+
+    @Extension
+    public static final class DescriptorImpl extends StepDescriptor {
+        @Override
+        public String getFunctionName() {
+            return "throttle";
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Throttle execution of node blocks within this body";
+        }
+
+        @Override
+        public boolean takesImplicitBlockArgument() {
+            return true;
+        }
+
+        @Override
+        public Set<? extends Class<?>> getRequiredContext() {
+            return Collections.singleton(TaskListener.class);
+        }
+
+        public FormValidation doCheckCategoryName(@QueryParameter String value) {
```

Comment:
Good question. This doesn't actually expose any meaningful information - all you can do is tell whether there's a category with a given name, not anything about the category, etc. I think it's harmless, but defer to others.